torch/csrc/jit/script/compiler.cpp

```diff
@@ -223,6 +223,26 @@ struct Environment {
       }
     }
   }
+
+  // for each captured input, create/refer new one in enclosing
+  // scope, and replace references in the block, delete branch block
+  // inputs, and the corresponding field in value_map
+  void deleteIfCapturedInputs(const SourceRange& loc) {
+    // block inputs should be 1:1 to the captured_inputs
+    JIT_ASSERT(b->inputs().size() == captured_inputs.size());
+    for(int i = captured_inputs.size() - 1; i >= 0; -- i) {
+      auto name = captured_inputs[i];
+      auto ident = Ident::create(loc, name);
+      auto parent_frame_var = next->getVar(ident);
+      auto captured_val = b->inputs()[i];
+      captured_val->replaceAllUsesWith(parent_frame_var);
+      if (value_table[name]->asValue(loc, method) == captured_val) {
```

Comment:
Would be good to add a comment explicitly calling out that it's safe to call `asValue` here since captured values are all guaranteed to be Simple